plan/logical_plan_builder.go

```diff
@@ -992,18 +1001,295 @@ func (g *gbyResolver) Leave(inNode ast.Node) (ast.Node, bool) {
+			if lCol == nil || rCol == nil {
+				continue
+			}
+			lTbl := tblInfoFromCol(x.Left, lCol)
+			if lTbl == nil {
+				lCol, rCol = rCol, lCol
+			}
+			switch x.Tp {
+			case ast.CrossJoin:
+				colDependMap[lCol] = rCol
+				colDependMap[rCol] = lCol
+			case ast.LeftJoin:
+				colDependMap[rCol] = lCol
+			case ast.RightJoin:
+				colDependMap[lCol] = rCol
+			}
+		}
+		return colDependMap
+	default:
+		return nil
+	}
+}
+
+func checkColFuncDepend(p LogicalPlan, col *expression.Column, tblInfo *model.TableInfo, gbyCols map[*expression.Column]struct{}, whereDepends, joinDepends map[*expression.Column]*expression.Column) bool {
+	for _, index := range tblInfo.Indices {
+		if !index.Unique {
+			continue
+		}
+		funcDepend := true
+		for _, indexCol := range index.Columns {
+			iColInfo := tblInfo.Columns[indexCol.Offset]
+			if !mysql.HasNotNullFlag(iColInfo.Flag) {
+				funcDepend = false
+				break
+			}
+			cn := &ast.ColumnName{
+				Schema: col.DBName,
+				Table:  col.TblName,
+				Name:   iColInfo.Name,
+			}
+			iCol, err := p.Schema().FindColumn(cn)
+			if err != nil {
+				log.Errorf("Can't find column %s.%s.%s from schema %s.", cn.Schema.L, cn.Table.L, cn.Name.L, p.Schema())
+				funcDepend = false
+				break
+			}
+			if _, ok := gbyCols[iCol]; ok {
+				continue
+			}
+			if wCol, ok := whereDepends[iCol]; ok {
+				if _, ok = gbyCols[wCol]; ok {
+					continue
+				}
+			}
+			if jCol, ok := joinDepends[iCol]; ok {
+				if _, ok = gbyCols[jCol]; ok {
+					continue
+				}
+			}
+			funcDepend = false
+			break
+		}
+		if funcDepend {
+			return true
 		}
-		g.err = errors.Errorf("Unknown column '%d' in 'group statement'", v.N)
+	}
+	primaryFuncDepend := true
+	hasPrimaryField := false
+	for _, colInfo := range tblInfo.Columns {
+		if !mysql.HasPriKeyFlag(colInfo.Flag) {
+			continue
+		}
+		hasPrimaryField = true
+		pCol, err := p.Schema().FindColumn(&ast.ColumnName{
+			Schema: col.DBName,
+			Table:  col.TblName,
+			Name:   colInfo.Name,
+		})
+		if err != nil {
+			primaryFuncDepend = false
+			break
+		}
+		if _, ok := gbyCols[pCol]; ok {
+			continue
+		}
+		if wCol, ok := whereDepends[pCol]; ok {
+			if _, ok = gbyCols[wCol]; ok {
+				continue
+			}
+		}
+		if jCol, ok := joinDepends[pCol]; ok {
+			if _, ok = gbyCols[jCol]; ok {
+				continue
+			}
+		}
+		primaryFuncDepend = false
+		break
+	}
+	return primaryFuncDepend && hasPrimaryField
+}
+
+func (b *planBuilder) checkOnlyFullGroupBy(p LogicalPlan, fields []*ast.SelectField, gby *ast.GroupByClause, from ast.ResultSetNode, where ast.ExprNode) {
+	gbyCols := make(map[*expression.Column]struct{}, len(fields))
+	gbyExprs := make([]ast.ExprNode, 0, len(fields))
+	schema := p.Schema()
+	for _, byItem := range gby.Items {
+		if colExpr, ok := byItem.Expr.(*ast.ColumnNameExpr); ok {
+			col, err := schema.FindColumn(colExpr.Name)
+			if err != nil {
+				log.Errorf("Can't find column %s from schema %s.", colExpr.Name, p.Schema())
```

Comment:
The error is not in the log.
`FindColumn` returns error when the column is ambiguous.
When column cannot be found it returns nil.

Reply:
I think we don't need to log the error, it will be returned later when resolving the column.